src/Curl.elm

```diff
@@ -0,0 +1,132 @@
+module Curl exposing (runCurl)
+
+import Cli.Option as Option
+import Cli.OptionsParser as OptionsParser exposing (OptionsParser)
+import Cli.OptionsParser.BuilderState
+import Cli.OptionsParser.MatchResult exposing (MatchResult(..))
+import Dict
+import InterpolatedField
+import Regex exposing (Regex)
+import Request exposing (Request)
+
+
+argsRegex : Regex
+argsRegex =
+    -- \s+|\s*'([^']*)'|\s*"([^"]*)"
+    --"\\s+|\\s*'([^']*)'|\\s*\"([^\"]*)\""
+    "\\s+|\\s*'([^']*)'|\\s*\"([^\"]*)\"|(\\S+)"
```

Comment:
Yeah, bash does pre-processing before arguments are passed to the actual script to execute. So if you run `echo "Hello there" how are you`, then it will break that down into 4 arguments, 1 for the quoted part, and 1 for each individual word after that.

So elm-cli-options-parser doesn't know how to do this pre-processing since it is usually used in a way where the pre-processing has already occurred, in the context of a node command. When you get the arguments in the context of a node script, that pre-processing has already occurred. So since this is in pure Elm, we need to do the equivalent of splitting out each argument and grouping together parts that are grouped by single- or double-quoted strings.